src/TeamStore.DevUI/Factories/AssetFactory.cs

```diff
@@ -0,0 +1,31 @@
+﻿namespace TeamStore.Factories
+{
+    using System.Collections.Generic;
+    using TeamStore.DevUI.ViewModels;
+    using TeamStore.Keeper.Models;
+
+    // These method names need to be renamed to follow a consistent pattern
+    public static class AssetFactory
+    {
+        public static List<AssetSearchViewModel> ConvertAssetSearch(List<Asset> assets)
+        {
+            var result = new List<AssetSearchViewModel>(assets.Count);
```

Comment:
Defensive programming -> NULL check on assets. To correct this:
- first create a Unit Test which fails, then create one that accepts an exception
- then fix the code so the test passes